src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -204,7 +204,7 @@ class FunctionBuilder[F >: Null](parameterTypeInfo: Array[MaybeGenericTypeInfo[_
 
   cn.interfaces.asInstanceOf[java.util.List[String]].add(interfaceTi.iname)
 
-  def result(): () => F = {
+  def result(initf: () => Unit = () => {}): () => F = {
```

Comment:
I would just call this `init`, the fact that its a function follows from its type.